packages/rum-core/src/domain/rumEventsCollection/action/trackClickActions.ts

```diff
@@ -0,0 +1,229 @@
+import type { Duration, ClocksState, RelativeTime, TimeStamp, Subscription } from '@datadog/browser-core'
+import {
+  Observable,
+  assign,
+  isExperimentalFeatureEnabled,
+  getRelativeTime,
+  ONE_MINUTE,
+  ContextHistory,
+  addEventListener,
+  DOM_EVENT,
+  generateUUID,
+  clocksNow,
+  ONE_SECOND,
+  elapsed,
+} from '@datadog/browser-core'
+import { FrustrationType, ActionType } from '../../../rawRumEvent.types'
+import type { RumConfiguration } from '../../configuration'
+import type { LifeCycle } from '../../lifeCycle'
+import { LifeCycleEventType } from '../../lifeCycle'
+import { trackEventCounts } from '../../trackEventCounts'
+import { waitIdlePage } from '../../waitIdlePage'
+import { getActionNameFromElement } from './getActionNameFromElement'
+
+interface ActionCounts {
+  errorCount: number
+  longTaskCount: number
+  resourceCount: number
+}
+
+export interface ClickAction {
+  type: ActionType.CLICK
+  id: string
+  name: string
+  startClocks: ClocksState
+  duration?: Duration
+  counts: ActionCounts
+  event: Event
+  frustrationTypes: FrustrationType[]
+}
+
+export interface ActionContexts {
+  findActionId: (startTime?: RelativeTime) => string | string[] | undefined
+}
+
+// Maximum duration for click actions
+export const CLICK_ACTION_MAX_DURATION = 10 * ONE_SECOND
+export const ACTION_CONTEXT_TIME_OUT_DELAY = 5 * ONE_MINUTE // arbitrary
+
+export function trackClickActions(
+  lifeCycle: LifeCycle,
+  domMutationObservable: Observable<void>,
+  { actionNameAttribute }: RumConfiguration
+) {
+  // TODO: this will be changed when we introduce a proper initialization parameter for it
+  const collectFrustrations = isExperimentalFeatureEnabled('frustration-signals')
+  const history = new ContextHistory<string>(ACTION_CONTEXT_TIME_OUT_DELAY)
+  const stopObservable = new Observable<void>()
+
+  lifeCycle.subscribe(LifeCycleEventType.SESSION_RENEWED, () => {
+    history.reset()
+  })
+
+  const { stop: stopListener } = listenClickEvents(onClick)
+
+  const actionContexts: ActionContexts = {
+    findActionId: (startTime?: RelativeTime) =>
+      isExperimentalFeatureEnabled('frustration-signals') ? history.findAll(startTime) : history.find(startTime),
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏
```suggestion
      collectFrustrations ? history.findAll(startTime) : history.find(startTime),
```